lib/puppet/resource_api.rb

```diff
@@ -262,17 +269,21 @@ def call_provider(value); end
         # force autoloading of the provider
         provider(name)
         attr_def = {}
-        my_provider.get(context).map do |resource_hash|
-          resource_hash.each do |key|
-            property = definition[:attributes][key.first]
-            attr_def[key.first] = property
-          end
-          context.type.namevars.each do |namevar|
-            if resource_hash[namevar].nil?
-              raise Puppet::ResourceError, "`#{name}.get` did not return a value for the `#{namevar}` namevar attribute"
+        begin
+          my_provider.get(context).map do |resource_hash|
+            resource_hash.each do |key|
+              property = definition[:attributes][key.first]
+              attr_def[key.first] = property
+            end
+            context.type.namevars.each do |namevar|
+              if resource_hash[namevar].nil?
+                raise Puppet::ResourceError, "`#{name}.get` did not return a value for the `#{namevar}` namevar attribute"
+              end
             end
+            Puppet::ResourceApi::TypeShim.new(resource_hash, name, context.type.namevars, attr_def)
           end
-          Puppet::ResourceApi::TypeShim.new(resource_hash, name, context.type.namevars, attr_def)
+        rescue NoMethodError => e
```

Comment:
* This might hide/modify errors from the `get()` call itself. 
* This is relevant to PDK-917, but the commit is not tagged
* There are many more calls of `get()` relevant to PDK-917 that need checking
* I've updated pdk-917 to be more specific about what has to be checked

Reply:
Have removed this commit from the PR, will work on separately for PDK-917